### PR TITLE
Event table fixes

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR updates the event table to have consistent timestamp values and adds the executed_epoch field.

Additionally, it adds postgres unit tests for the ScabbardStore to start.  These are enabled with the experimental feature `"diesel-postgres-tests"` and actually executed with the existence of the `DIESEL_POSTGRES_TEST_URL` environment variable.

### CODE REVIEW

**Review by isabeltomb — DISMISSED** [review]

File: services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    id                        INTEGER PRIMARY KEY AUTOINCREMENT,
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    event_type                TEXT NOT NULL
+        CHECK ( event_type IN ('ALARM', 'DELIVER', 'START', 'VOTE') ),
+    created_at                TEXT
+        DEFAULT (strftime('%Y-%m-%d %H:%M:%f', 'now', 'localtime')) NOT NULL,
+    executed_at               TEXT,
+    executed_epoch            BIGINT,
+    FOREIGN KEY (circuit_id, service_id) REFERENCES scabbard_service(circuit_id, service_id)
```
Comment (agunde406) [inline services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql:48]: This migration needs to have ON DELETE CASCADE so the event will be properly cleaned up when scabbard service is purged.

File: services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+);
+
+INSERT INTO consensus_2pc_event
+SELECT t.id, t.circuit_id, t.service_id, t.event_type, t.created_at, t.executed_at,
+       CASE t.executed_at
+           WHEN NOT NULL THEN ctx.epoch
+           ELSE NULL
+       END as executed_epoch
+FROM temp_consensus_2pc_event t,
```
Comment (agunde406) [inline services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql:57]: The temp_consensus_2pc_event table is never dropped

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql (hunk 1 above)
Comment (isabeltomb) [reply services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql]: Added ON DELETE CASCADE

File: services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    service_id                TEXT NOT NULL,
+    event_type                TEXT NOT NULL,
+    created_at                TEXT NOT NULL,
+    executed_at               TEXT
+);
+
+INSERT INTO temp_consensus_2pc_event
+SELECT id, circuit_id, service_id, event_type,
+       strftime('%Y-%m-%d %H:%M:%f', created_at, 'localtime') as created_at,
+       strftime('%Y-%m-%d %H:%M:%f', executed_at, "unixepoch", "localtime") as executed_at
```
[the remark this answers is not in the document]
Comment (isabeltomb) [reply services/scabbard/libscabbard/src/migrations/diesel/sqlite/migrations/2022-07-05-014045-event-table-updates/up.sql:33]: added a commit with a trigger that will fix this after an update or an insert

**Review by vaporos — APPROVED** [review]

**Pull request merged**